lib/signup/sign_up_screen.dart

```diff
@@ -127,14 +162,64 @@ class _HomeContent extends StatelessWidget {
                           TextStyle(fontWeight: FontWeight.w600, fontSize: 12)),
                   IconTextField(
                       imagePath: Assets.assest.images.user.path,
-                      hintText: "Name"),
+                      hintText: "Name",
+                      onTextChange: (value) {
+                        setNameTextState(value);
+                      },
+                      ),
+                      
                   IconTextField(
                       imagePath: Assets.assest.images.email.path,
-                      hintText: "Email"),
+                      hintText: "Email",
+                      onTextChange: (value) {
+                        setEmailTextState(value);
+                      },
+                      ),
                   IconTextField(
                       imagePath: Assets.assest.images.password.path,
-                      hintText: "Password"),
-                  TextButton(onPressed: () {}, child: const Text("Signup")),
+                      hintText: "Password",
+                      onTextChange:(value) {
+                        setPasswordState(value);
+                      },),
+                    BlocSelector<SignUpCubit, SignUpState, String?>(
+                      selector: (state) {
+                          if (kDebugMode) {
+                           print('$state');
+                         }
+                        if(state is SignUpFailedState)
+                        {
+                          return state.provider;
+                        }
+                        return null;
+                      },
+                      builder: (context, state) {
+                        
+                        if(state == "Name" || state =="Email" || state == "Password")
+                        {
+                          return Text("$state must not be empty",
+                            style: const TextStyle(color:Colors.red),
+                          );
+                        }
+                        return const SizedBox.shrink();
+                      },
+                    ),
+                  TextButton(onPressed: () {
+                    if(textName.isEmpty){
```

Comment:
there's no UI update needed for textEmail, textPassword, textName, so please do not use them and add setState() for them.

Refer to my answer here: https://github.com/PhongDavidVu/MentorME-Batch16/pull/1#discussion_r1613165364

Contact David to give you access to his repo if it's private repo

Reply:
I intended to to like that, but was sure how to get the state from cubit

Reply:
Rivised the code